src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -559,6 +559,13 @@ private async Task InitPeer()
         NodeRecordSigner nodeRecordSigner = new(_api.EthereumEcdsa, new PrivateKeyGenerator().Generate());
         EnrRecordParser enrRecordParser = new(nodeRecordSigner);
         EnrDiscovery enrDiscovery = new(enrRecordParser, _api.LogManager); // initialize with a proper network
+
+        if (_networkConfig.DisableDiscV4DnsFeeder)
+        {
+            // Feed some nodes into discoveryApp in case all bootnodes is faulty.
+            _api.DisposeStack.Push(new NodeSourceToDiscV4Feeder(enrDiscovery, _api.DiscoveryApp, 50));
```

Comment:
@asdacap Looking at the comment this logic should be triggered when all bootnodes are faulty - I don't see that check and it seems actually to be nice - also why are we checking bootnodes only? How about trusted nodes?

Reply:
DNS discovery runs in parallel with DiscV4, so if I were to check it first, I'd need to make a more intrusive change otherwise. By "trusted" in the log, it just check node that have responded. There aren't really a "trusted" flag.